packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -467,27 +473,20 @@ export function usePress(props: PressHookProps): PressResult {
         }
 
         // Only handle left clicks
-        // Safari on iOS sometimes fires pointerup events, even
-        // when the touch isn't over the target, so double check.
-        if (e.button === 0 && isOverTarget(e, e.currentTarget)) {
+        if (e.button === 0) {
```

Comment:
I am not certain yet if this bug still exists in Safari. This was added in #335. Will need to test.